app/Models/User.php

```diff
@@ -495,21 +496,57 @@ public static function lookupWithHistory($usernameOrId, $type = null, $findAll =
         }
     }
 
-    public function addToGroup(Group $group, ?self $actor = null): void
+    public function addToGroup(Group $group, ?array $playmodes = null, ?self $actor = null): void
```

Comment:
not really important but looking at the length of this file, this function could be in `UserGroup` instead.

Reply:
i think fits better in this class, or at least to me the mental model is more like "add user to group" than "make a new usergroup"